### PR TITLE
Add Resource Registration Component.

### PR DESCRIPTION
**Pull request by isuri97**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,26 +1,68 @@
...
+.idea/dictionaries
+
+# Sensitive or high-churn files:
+.idea/**/dataSources/
+.idea/**/dataSources.ids
+.idea/**/dataSources.xml
+.idea/**/dataSources.local.xml
+.idea/**/sqlDataSources.xml
+.idea/**/dynamic.xml
+.idea/**/uiDesigner.xml
```
Comment (JKAUSHALYA) [inline .gitignore]: No need to include each and every file related to .idea. We can completely ignore it.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/exceptions/ResourceEndpointException.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+public class ResourceEndpointException extends WebApplicationException {
+
+    public ResourceEndpointException(Response.Status status, ErrorDTO errorDTO) {
+
+        super(Response.status(status)
+                .entity(errorDTO)
+                .build());
+    }
+
+    public ResourceEndpointException(Response.Status status) {
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/exceptions/ResourceEndpointException.java]: Need to add doc comments for all public methods (Can ignore getters and setters if you want)

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+    private static final Log log = LogFactory.getLog(ResourceRegistrationApiServiceImpl.class);
+
+    /**
+     * Register a resource with resource details
+     *
+     * @param requestedResource details of the resource to be registered
+     * @return Response with the status of the registration
+     */
+
+    @Override
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java]: No need a new line here as well. Please correct all occurrences.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+                    Resource resourceRegistration = ResourceUtils.getResourceService().getResourceById
+                            (resourceId);
+                    ReadResourceDTO readResourceDTO = ResourceUtils.readResponse(resourceRegistration);
+                    return Response.status(Response.Status.OK).entity(readResourceDTO).build();
+                } catch (UMAServiceException e) {
+                    handleErrorResponse(e, log);
+                } catch (Throwable throwable) {
+                    log.error("Internal server error occurred. ", throwable);
+                }
+
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java]: Unnecessary new line ?

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+        // For testing purpose current resource owner id is taken to validate pat access token
+        resourceOwnerId = "123";
+        try {
+            List<String> resourceRegistration = ResourceUtils.getResourceService().getResourceIds(resourceOwnerId);
+            Response response = Response.ok().entity(resourceRegistration).build();
+            return response;
+        } catch (UMAClientException e) {
+            handleErrorResponse(e, log);
+            log.error("Invalid request.Request with valid resource Id to update the resource. ", e);
+            //return Response.status(status).entity(errorDTO).build();
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java]: What is this comment ?

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+    /**
+     * Retrieve the available resourceId list
+     *
+     * @param resourceOwnerId
+     * @return Response with the retrieved resourceId's/ retrieval status
+     */
+    @Override
+    public Response getResourceIds(String resourceOwnerId) {
+
+        // For testing purpose current resource owner id is taken to validate pat access token
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java]: When commenting start with caps and end with a full stop. Please follow that pattern for all comments.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,247 @@
...
+        }
+        return null;
+    }
+
+    /**
+     * Update a resource
+     *
+     * @param updatedResource details of the resource to be updated
+     * @param resourceId      ID of the resource to be updated
+     * @return
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java]: You need to fill all the details when adding a doc comment. Please fix all the occurrences.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+        } catch (UMAClientException e) {
+            handleErrorResponse(e, log);
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when retrieving resource from the resource server.", e);
+            }
+        }
+        return null;
+    }
+
+    public static void handleErrorResponse(UMAException umaException, Log log) throws ResourceEndpointException {
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImpl.java]: Need doc comments. Please fix all occurrences.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/util/ResourceUtils.java [hunk 9]
```diff
@@ -0,0 +1,119 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/util/ResourceUtils.java]: Year ?

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/resourceregistration.yaml [hunk 10]
```diff
@@ -0,0 +1,302 @@
+swagger: '2.0'
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/resources/resourceregistration.yaml]: If we put this, we may need to add license header for this as well.

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql [hunk 11]
```diff
@@ -0,0 +1,100 @@
+CREATE TABLE IDN_RESOURCE_SET
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql]: Change the file name to reflect the operation and which DBMS.

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql [hunk 12]
```diff
@@ -0,0 +1,100 @@
...
+(
+RESOURCE_SET_ID varchar(30) PRIMARY KEY,
+RESOURCE_SET_NAME VARCHAR (30),
+TIME_CREATED timestamp NOT NULL,
+TOKEN_ID VARCHAR NOT NULL,
+RESOUCE_OWNER_ID varchar(30),
+RESOURCE_SERVER_ID varchar(30),
+RESOURCE_OWNER_NAME varchar(30),
+ACCESS_TOKEN VARCHAR(20)
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql]: Can we have indentation for SQL as well ?

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql [hunk 13]
```diff
@@ -0,0 +1,100 @@
...
+SCOPE_ID varchar(30) PRIMARY KEY,
+RESOURCE_SET_ID varchar(30),
+PROPERTY_KEY varchar(40),
+PROPERTY_VALUE varchar(30),
+CONSTRAINT FK_Resource_Meta_Data FOREIGN KEY (RESOURCE_SET_ID) REFERENCES IDN_RESOURCE_SET (RESOURCE_SET_ID)
+);
+
+CREATE TABLE IDN_RESOURCE_SET_SCOPE
+(
+SCOPE_ID varchar(30) PRIMARY KEY,
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql]: Don't use upper and lower case. Use upper for keywords and lower for rest.

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql [hunk 14]
```diff
@@ -0,0 +1,100 @@
...
+  ID                INTEGER AUTO_INCREMENT PRIMARY KEY NOT NULL,
+  RESOURCE_ID         VARCHAR(255),
+  PROPERTY_KEY      VARCHAR(255),
+  PROPERTY_VALUE    VARCHAR(255),
+  CONSTRAINT FK_RESOURCE_META_DATA FOREIGN KEY (RESOURCE_ID) REFERENCES IDN_RESOURCE (RESOURCE_ID)
+  ON DELETE CASCADE
+);
+
+/*CREATE TABLE IDN_RESOURCE_SCOPE (
+  ID          INTEGER AUTO_INCREMENT PRIMARY KEY NOT NULL,
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/resources/tables.sql]: Why commented ?

File: components/identity.oauth.uma.resource.endpoint/src/main/webapp/WEB-INF/beans.xml [hunk 15]
```diff
@@ -0,0 +1,17 @@
...
+    <jaxrs:server id="services" address="/">
+        <jaxrs:serviceBeans>
+            <bean class="org.wso2.carbon.identity.oauth.uma.endpoint.ResourceRegistrationApi"/>
+            
+        </jaxrs:serviceBeans>
+        <jaxrs:providers>
+            <bean class="com.fasterxml.jackson.jaxrs.json.JacksonJsonProvider"/>
+        </jaxrs:providers>
+    </jaxrs:server>
+</beans>
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/webapp/WEB-INF/beans.xml]: New line.

File: components/identity.oauth.uma.resource.endpoint/src/main/webapp/WEB-INF/beans.xml [hunk 16]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/webapp/WEB-INF/beans.xml]: Needs license header.

File: components/identity.oauth.uma.resource.endpoint/src/test/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImplTest.java [hunk 17]
```diff
@@ -0,0 +1,149 @@
+/*
+ *  Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/test/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImplTest.java]: Year ?

File: components/identity.oauth.uma.resource.endpoint/src/main/webapp/WEB-INF/web.xml [hunk 18]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/main/webapp/WEB-INF/web.xml]: License header.

File: components/identity.oauth.uma.resource.endpoint/src/test/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImplTest.java [hunk 19]
```diff
@@ -0,0 +1,149 @@
...
+        }
+
+    }
+
+    @Test
+    public void testGetResourceIds() throws Exception {
+
+        List<String> resourceOwnerIds = new ArrayList<>();
+/*
+        when(resourceService.getResourceIds("123")).thenThrow(new UMAServiceException("This is a server exception"));
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/test/java/org/wso2/carbon/identity/oauth/uma/endpoint/impl/impl/ResourceRegistrationApiServiceImplTest.java]: Why commented ?

File: components/identity.oauth.uma.resource.endpoint/src/test/resources/testng.xml [hunk 20]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!--
+  ~ Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.endpoint/src/test/resources/testng.xml]: Year ?

File: components/identity.oauth.uma.resource.service/pom.xml [hunk 21]
```diff
@@ -0,0 +1,224 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (JKAUSHALYA) [inline components/identity.oauth.uma.resource.service/pom.xml]: License.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/ResourceEndpointConstants.java [hunk 22]
```diff
@@ -0,0 +1,53 @@
...
+
+package org.wso2.carbon.identity.oauth.uma.resource.endpoint;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * ResourceEndpointConstants class is used to handle exceptions in the endpoint layer.
+ * */
+public class ResourceEndpointConstants {
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/ResourceEndpointConstants.java]: Need to change the name of this class

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/ResourceEndpointConstants.java [hunk 23]
```diff
@@ -0,0 +1,53 @@
...
+
+    private String code;
+    private Map<String, String[]> map;
+
+    public ResourceEndpointConstants() {
+
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/ResourceEndpointConstants.java]: remove unwanted empty lines.
What does this code mean?

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 24]
```diff
@@ -0,0 +1,251 @@
...
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
+            }
+            handleErrorResponse(e, log);
+        } catch (Throwable throwable) {
+            log.error("Internal server error occurred. ", throwable);
+        }
+        return null;
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Why do we return null?

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 25]
```diff
@@ -0,0 +1,251 @@
...
+            return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+        } catch (UMAServiceException e) {
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
+            }
+            handleErrorResponse(e, log);
+        } catch (Throwable throwable) {
+            log.error("Internal server error occurred. ", throwable);
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Don't we need to return an error message?

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 26]
```diff
@@ -0,0 +1,251 @@
...
+        try {
+            Resource registerResource = ResourceUtils.getResourceService()
+                    .registerResource(ResourceUtils.getResource(requestedResource));
+            CreateResourceDTO createResourceDTO = ResourceUtils.createResponse(registerResource);
+            return Response.status(Response.Status.CREATED).entity(createResourceDTO).build();
+        } catch (UMAServiceException e) {
+            handleErrorResponse(e, log);
+        } catch (UMAClientException e) {
+            if (log.isDebugEnabled()) {
+                log.debug("Client error when registering resource in resource server.", e);
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Are we doing a special thing in this debug log? seems to be we log the same in error handling.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 27]
```diff
@@ -0,0 +1,251 @@
...
+        ErrorDTO errorDTO = new ErrorDTO();
+        errorDTO.setCode(code);
+        errorDTO.setDescription(description);
+        return new ResourceEndpointException(status, errorDTO);
+    }
+
+    /**
+     * Validate resourceId
+     *
+     * @param resourceId resourceId of the resource which need to get deleted
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Is this only for deleting a resource?

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 28]
```diff
@@ -0,0 +1,251 @@
...
+     * @return validator obtain
+     */
+    public boolean isResourceId(String resourceId) throws UMAClientException {
+
+        boolean validator = true;
+        String pattern = "^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$";
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Better to check with the valid pattern.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java [hunk 29]
```diff
@@ -0,0 +1,251 @@
...
+        Pattern pat = Pattern.compile(pattern);
+        Matcher match = pat.matcher(resourceId);
+
+        if (match.find()) {
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: You can directly return the result for match.find()

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/util/ResourceUtils.java [hunk 30]
```diff
@@ -0,0 +1,116 @@
...
+     * @return A generic resourceregistration with the specified details
+     */
+    public static Resource getResource(ResourceDetailsDTO resourceDetailsDTO) {
+
+        Resource resourceRegistration = new Resource();
+        resourceRegistration.setName(resourceDetailsDTO.getName());
+        resourceRegistration.setScopes(resourceDetailsDTO.getResource_scopes());
+
+        for (String scope : resourceRegistration.getScopes()) {
+            resourceRegistration.getScopeDataDOArr().add(new ScopeDataDO(resourceRegistration.getResourceId(), scope));
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/util/ResourceUtils.java]: What is this getScopeDataDO**Arr**?

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/uma_resource.sql [hunk 31]
```diff
@@ -0,0 +1,36 @@
+
+CREATE TABLE IDN_RESOURCE (
```
Comment (isharak) [inline components/identity.oauth.uma.resource.endpoint/src/main/resources/uma_resource.sql]: Need to define primary key, unique keys, indexes etc.

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/resourceregistration.yaml (hunk 10 above)
Comment (isuri97) [reply components/identity.oauth.uma.resource.endpoint/src/main/resources/resourceregistration.yaml]: Done.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java (hunk 25 above)
Comment (isuri97) [reply components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Returned error message.

File: components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java (hunk 29 above)
Comment (isuri97) [reply components/identity.oauth.uma.resource.endpoint/src/main/java/org/wso2/carbon/identity/oauth/uma/resource/endpoint/impl/ResourceRegistrationApiServiceImpl.java]: Done.

File: components/identity.oauth.uma.resource.endpoint/src/main/resources/resourceregistration.yaml (hunk 10 above)
Comment (isuri97) [reply components/identity.oauth.uma.resource.endpoint/src/main/resources/resourceregistration.yaml]: Done.

File: components/identity.oauth.uma.resource.endpoint/pom.xml [hunk 32]
```diff
@@ -0,0 +1,258 @@
...
+                        </resource>
+                    </webResources>
+                    <warName>identity.oauth.uma.resource.endpoint</warName>
+                </configuration>
+            </plugin>
+            <!-- Plugin for generating the swagger server stubs-->
+            <plugin>
+                <groupId>org.wso2.maven.plugins</groupId>
+                <artifactId>swagger2cxf-maven-plugin</artifactId>
+                <version>1.0-SNAPSHOT</version>
```
[the remark this answers is not in the document]
Comment (isuri97) [reply components/identity.oauth.uma.resource.endpoint/pom.xml]: This dependency used only for generating swagger file . And it remains as commented dependency in endpoint pom.xml .

**Pull request merged**